aws/data_source_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsElasticacheReplicationGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsElasticacheReplicationGroupRead,
+		Schema: map[string]*schema.Schema{
+			"replication_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replication_group_description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"auth_token_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"automatic_failover_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"configuration_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"primary_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"number_cache_clusters": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"node_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_window": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"snapshot_retention_limit": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+	input := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupId: aws.String(d.Get("replication_group_id").(string)),
+	}
+
+	resp, err := conn.DescribeReplicationGroups(input)
+	if err != nil {
+		return errwrap.Wrapf("Error retrieving Elasticache Replication Group: {{err}}", err)
```

Comment:
Nitpick: The errwrap IMO isn't necessary in this context as there's no code that would need to know the original error above CRUD. Pretty much all errors from CRUD end up as messages printed in the CLI, so there's not much value in using this over simple `fmt.Errorf()`.